src/AutoRest.CSharp/Common/Input/InputTypes/InputIntrinsicType.cs

```diff
@@ -3,10 +3,8 @@
 
 namespace AutoRest.CSharp.Common.Input;
 
-internal record InputIntrinsicType(InputIntrinsicTypeKind Kind) : InputType(InputIntrinsicTypeName, false)
+internal record InputIntrinsicType(InputIntrinsicTypeKind Kind) : InputType(Kind.ToString(), false)
```

Comment:
maybe we should stick to `Kind`?

Reply:
It might be misleading.
This is following what we are doing in the emitter.
Kind discriminates the types into primitive, model, enum, etc.
Name discriminates primitive type into its own variants (which is not a real discriminator because we do not have derived types)

Reply:
For details, you could check the changes in `inputType.ts` in the emitter side.

Reply:
Also, to reflect the shape on the emitter side, we should have
```
internal record InputIntrinsicType(InputIntrinsicTypeKind Name) : InputType(Name.ToString(), false)
```
but unfortunately, C# does not support to inherit a class and change the type of one inherited property.
In the emitter, we have:
```
interface InputType {
Name: string;
Kind: string; // this will not show on the C# side
IsNullable: boolean;
}
interface InputIntrinsicType extends InputType {
Kind: "Intrinsic";
Name: InputIntrinsicTypeKind; // enums in ts is actually literal strings, therefore this is OK
}
```
but enums in C# is not strings we cannot do that.